homeassistant/components/elkm1/alarm_control_panel.py

```diff
@@ -38,9 +38,12 @@
     if discovery_info is None:
         return
 
-    elk = hass.data[ELK_DOMAIN]['elk']
-    entities = create_elk_entities(hass, elk.areas, 'area', ElkArea, [])
-    async_add_entities(entities, True)
+    elk_datas = hass.data[ELK_DOMAIN]
+    for _, elk_data in elk_datas.items():
+        elk = elk_data['elk']
+        entities = create_elk_entities(elk_data, elk.areas,
+                                       'area', ElkArea, [])
+        async_add_entities(entities, True)
```

Comment:
Collect entities in a list first, then make a single call to `async_add_entities`.